core/consensus/babe/src/lib.rs

```diff
@@ -463,12 +477,13 @@ fn check_header<B: Block + Sized, C: AuxStore>(
 		babe_err!("Header {:?} has a bad seal", hash)
 	})?;
 
-	let pre_digest = find_pre_digest::<B>(&header)?;
-	let BabePreDigest { slot_num, index, ref proof, ref vrf_output } = pre_digest;
+	let (_epoch, pre_digest) = find_pre_digest::<B>(&header)?;
+
+	let BabePreDigest { slot_num, index, ref proof, ref vrf_output, epoch: _ } = pre_digest;
 
 	if slot_num > slot_now {
 		header.digest_mut().push(seal);
-		Ok(CheckedHeader::Deferred(header, slot_num))
+		Ok(CheckedHeader::Deferred(([0; 32], header), slot_num))
```

Comment:
Do we maybe want to change `CheckedHeader` so accept a different type for `Deferred` and `Checked`?
Or at least return `Option<[u8;32}>` here?